scene/resources/visual_shader.compat.inc

```diff
@@ -0,0 +1,48 @@
+/**************************************************************************/
+/*  visual_shader.compat.inc                                              */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef DISABLE_DEPRECATED
+
+void VisualShaderNodeComment::_set_description_bind_compat_88014(const String &p_description) {
+	set_meta("description", p_description);
+}
+
+String VisualShaderNodeComment::_get_description_bind_compat_88014() const {
+	return get_meta("description", "");
+}
+
+void VisualShaderNodeComment::_bind_compatibility_methods() {
+	ClassDB::bind_compatibility_method(D_METHOD("set_description", "description"), &VisualShaderNodeComment::_set_description_bind_compat_88014);
+	ClassDB::bind_compatibility_method(D_METHOD("get_description"), &VisualShaderNodeComment::_get_description_bind_compat_88014);
+
+	ADD_PROPERTY(PropertyInfo(Variant::STRING, "description"), "set_description", "get_description");
+}
```

Comment:
Actually now that I think further about it:
- I think this can be "normal" method bindings in `_bind_methods` instead of compat bindings. The class itself is already deprecated/for compat so the methods themselves don't need to be too.
- If so, we can also re-add the `description` property fully, and document that it doesn't do anything in this new setup, but is used to preserve data authored in earlier Godot versions.

Reply:
To be clear, this means also moving them to `visual_shader.cpp` (enclosed in `#ifndef DISABLE_DEPRECATED`) and the `.compat.inc` file can be removed.

Reply:
Alright, done :)